### ISSUE TITLE
GACOS ERROR 

### ISSUE THREAD

**Issue by CUMT530849827** [esrivas17/gacos_correction_for_GMTSAR#5]
Hello, author, I would like to ask about the error I reported during the operation of GACOS correction. The following is the specific log of the error
error1:
rdtrack [WARNING]: Some input points were outside the grid domain(s).
grdtrack [WARNING]: Some input points were outside the grid domain(s).
grdmath [ERROR]: Operation "SUB" requires 2 operands
grdmath [ERROR]: Operation "SUB" requires 2 operands

error2:
surface [INFORMATION]: Writing grid to file lla.grd
grdtrack [WARNING]: Some input points were outside the grid domain(s).
grdtrack [WARNING]: Some input points were outside the grid domain(s).
grdmath [ERROR]: Operation "SUB" requires 2 operands
grdmath [ERROR]: Operation "SUB" requires 2 operands
Problems with the reference point. Is the reference point within the AOI?
GACOS correction done

![image](https://user-images.githubusercontent.com/49933485/205007343-d78f3ecf-eedf-4689-9972-a9b9509cf5e5.png)

![image](https://user-images.githubusercontent.com/49933485/205007450-044d8f95-45ed-4e79-a1ed-b4752366d952.png)
![image](https://user-images.githubusercontent.com/49933485/205007656-822f1ce3-5b18-4854-b309-54cbd28452a5.png)
![Uploading image.png…]()

**Comment by CUMT530849827** [comment]
ref_point
101.9389 37.4718

**Comment by esrivas17** [comment]
Are your GACOS grids slightly larger in longitude with respect to the DEM? Maybe trying with a larger DEM?

**Comment by Elcappo2019** [comment]
@esrivas17 , I faced the same problem, the limits of dem is more than the GACOS grid file !!! but still the same error , Can you help me to resolve it ? 
![error1](https://github.com/esrivas17/gacos_correction_for_GMTSAR/assets/45139381/d9d2e862-2a91-4bbe-be96-6736474fd723)
![demVsdate1_ztd](https://github.com/esrivas17/gacos_correction_for_GMTSAR/assets/45139381/97daf3c2-4744-45c7-b958-eb95c3ee8d0a)

**Comment by esrivas17** [comment]
Hiii. According to the log, it seems the problem comes from the reference point. Could you check it is in the right format and that it is within the AOI?

**Comment by Elcappo2019** [comment]
![ref_point](https://github.com/esrivas17/gacos_correction_for_GMTSAR/assets/45139381/b145dc88-e4ea-405c-a092-58cf127efbe5)

**Comment by Elcappo2019** [comment]
![ref_point](https://github.com/esrivas17/gacos_correction_for_GMTSAR/assets/45139381/3c326fd2-33dd-492d-a60c-762d9dd83a6f)

**Comment by esrivas17** [comment]
Ok. It seems that you are processing 3 subswaths. Originally, the scripts were not meant to work with more than 2 subswaths. I would need to add this feature to the code.
I think that it is crashing when reprojecting GACOS to radar coordinates.

**Comment by esrivas17** [comment]
Also, there are some inaccuracies when reprojecting the grid from geographic coordinates to radar. Meaning that when the limits are too close to each other (between Gacos and Dem) when reprojecting some points might fall outside of the grid. You could try with a new Dem that is slightly longer in the Y minimum boundary

**Comment by esrivas17** [comment]
> Also, there are some inaccuracies when reprojecting the grid from geographic coordinates to radar. Meaning that when the limits are too close to each other (between Gacos and Dem) when reprojecting some points might fall outside of the grid. You could try with a new Dem that is slightly longer in the Y minimum boundary 

You can better try first to run line 73 from operation.csh and to see if it runs successfully or not

**Comment by Elcappo2019** [comment]
Dear @erik-17 , I run the line 73 in the operation.csh in the top terminal. and i think is Ok (see the bottom terminal" .
![run173](https://github.com/esrivas17/gacos_correction_for_GMTSAR/assets/45139381/d7acd77b-78a5-4fb3-b09d-988236658de9)

**Comment by esrivas17** [comment]
> Dear @erik-17 , I run the line 73 in the operation.csh in the top terminal. and i think is Ok (see the bottom terminal" .
> ![run173](https://github.com/esrivas17/gacos_correction_for_GMTSAR/assets/45139381/d7acd77b-78a5-4fb3-b09d-988236658de9)
> 

Right. Looks fine. One other thing. Could you check the reference point is within phasefilt.grd ?
It looks like the Gacos grid dimensions are fine. 

Also, what script are you using? Single_gacos_correction.csh?

**Comment by Elcappo2019** [comment]
I confused, the limits of phasefilt_ll.grd appeared as in the first attached terminal 
![goog](https://github.com/esrivas17/gacos_correction_for_GMTSAR/assets/45139381/ba49e333-ee23-486e-8ae1-6daaab721b31)
i think the problem is the study area is located in the west of (Greenwich line)  with " minus coordinates " and i put the reference point as " -8.077282 31.504392 " but the limit of "  phasefilt_ll.grd  are "  x_min: 350.638888889 x_max: 353.588888889 " and " y_min: 30.2958333338 y_max: 32.3250000005" So, How to fix it ?

**Comment by Elcappo2019** [comment]
I used " operation.csh " script .

**Comment by Elcappo2019** [comment]
When i make a new phasefilt.grd file and make sure the coordinates with X in minus signs fitting the coordinate of ref_point. in addition, during this case i used  "Single_gacos_correction.csh " 
![tt](https://github.com/esrivas17/gacos_correction_for_GMTSAR/assets/45139381/ba8d4e86-a03b-44a7-b048-7e919f86fd15)
Still this error !!!! where is the problem ?

**Comment by esrivas17** [comment]
> When i make a new phasefilt.grd file and make sure the coordinates with X in minus signs fitting the coordinate of ref_point. in addition, during this case i used  "Single_gacos_correction.csh " 
> ![tt](https://github.com/esrivas17/gacos_correction_for_GMTSAR/assets/45139381/ba8d4e86-a03b-44a7-b048-7e919f86fd15)
> Still this error !!!! where is the problem ?

Mmm everything looks fine. Could you check again the info of resample_zpddm.grd? And ensure it has the same dimensions and incs of phasefilt_ll.grd?
If everything is fine you can check what output you get by running `gmt grdtrack` using your reference point to `phasefilt_ll.grd` and `resample_zpddm.grd`
It's indeed strange. I can't see an obvious mistake. But I have spotted some misleading information in the script
